### PR TITLE
Final changes to schema to match 4.2 data model IG

### PR DESCRIPTION
**Pull request by dehall**
These are the (hopefully) final changes to this schema to match the forthcoming version 4.2 of the Data Model IG. Changes are described in ancillary_codi_tables.sql, plus the CENSUS_DEMOG table has been removed following discussions with the implementation teams

### CODE REVIEW

File: schema_vdw.sql [hunk 1]
```diff
@@ -1,115 +1,29 @@
...
+	
+*/
+
+
 CREATE SCHEMA VDW;
 
 CREATE TABLE VDW.CENSUS_DEMOG (
   CENSUS_YEAR int NOT NULL,
   GEOCODE varchar(15) NOT NULL,
-  BLOCK varchar(3) NULL,
```
Comment (radamson) [inline schema_vdw.sql:6]: These are valid columns according to the [CODI Data Models IG v4.1.5 (See Table 12 on page 46)](https://raw.githubusercontent.com/mitre/codi/main/CODI%20Data%20Model%20Implementation%20Guide.pdf)

**Comment by radamson** [comment]
Based on some slack discussions with Catherine it looks like these are CODI DM 4.2 changes. Can you update the title and description to reflect that?

File: ancillary_codi_tables.sql [hunk 2]
```diff
@@ -168,7 +185,7 @@ CREATE TABLE CODI.REFERRAL
 	CHECK(REFERRAL_STATUS in ('A', 'D', 'NI', 'UN', 'OT')),
 	CHECK(REFERRAL_PRIOR_AUTH in ('Y', 'N', 'R', 'NI', 'UN', 'OT')),
 	PRIMARY KEY(REFERRALID),
-	FOREIGN KEY(SOURCE_PROVIDER_ID) REFERENCES CDM.PROVIDER (PROVIDERID),
+	FOREIGN KEY(SOURCE_PROVIDERID) REFERENCES CDM.PROVIDER (PROVIDERID),
```
Comment (radamson) [inline ancillary_codi_tables.sql:188]: `SOURCE_PROVIDER_ID` should be changed to `SOURCE_PROVIDERID` on line 181 as well (I can't comment on that line specifically).

File: schema_vdw.sql (hunk 1 above)
Comment (radamson) [reply schema_vdw.sql:6]: Should CENSUS_DEMOG just be totally removed? Looks like the draft CODI 4.2 DM IG removes this table.

Comment (dehall) [reply schema_vdw.sql:6]: Yeah it's been commented out now but we do of course have git history so should probably delete it instead

File: ancillary_codi_tables.sql [hunk 3]
```diff
@@ -413,7 +433,9 @@ CREATE TABLE CDM.PRIVATE_ADDRESS_HISTORY
 	ADDRESS_STATE char(2) NULL,
 	ADDRESS_TYPE char(2) NOT NULL,
 	ADDRESS_PREFERRED char(2) NOT NULL,
-	ADDRESS_USE char(2) NULL,
+	ADDRESS_PERIOD_END date NULL,
+	ADDRESS_PERIOD_START date NULL,
+	ADDRESS_USE char(2) NOT NULL,
```
Comment (radamson) [inline ancillary_codi_tables.sql:438]: `ADDRESS_USE` is nullable in the draft CODI DM IG 4.2

Comment (radamson) [reply ancillary_codi_tables.sql:438]: Looks like it is required in the PCOR data model though.

**Review by radamson — APPROVED** [review]

**Pull request merged**